pkg/bee/api/files.go

```diff
@@ -14,7 +14,7 @@ import (
 type FilesService service
 
 // Download downloads data from the node
-func (f *FilesService) Download(ctx context.Context, a swarm.Address) (resp io.ReadCloser, err error) {
+func (f *FilesService) Download(ctx context.Context, a swarm.Address) ([]byte, error) {
```

Comment:
This can potentially lead to resource exhaustion with larger files. The ReadCloser is deliberately returned to signal that it has to be closed.

Reply:
The bug I had to dig deep to find was that the reader was not closed where this  method was used. We are not at the level of testing with large enough files that would lead to resource exhaustion, and returning bytes is more explicit.

 I don't mind adding the missing Close statement instead, either way works.

Reply:
I liked the explicit return of bytes cause the current implementation has to led a bugs in a couple of places.
Not sure if we are at the position of testing with files larger than a few MB.

Reply:
I see this as a regression to put the whole body in memory instead just to add a Close or drain to the returned io.ReadCloser.

Assuming the size of the body makes the implementation more fragile for the people in the future.